src/logic/function/basic.lean

```diff
@@ -79,6 +79,20 @@ def injective.decidable_eq [decidable_eq β] (I : injective f) : decidable_eq α
 lemma injective.of_comp {g : γ → α} (I : injective (f ∘ g)) : injective g :=
 λ x y h, I $ show f (g x) = f (g y), from congr_arg f h
 
+lemma injective.dite_injective (p : α → Prop) [decidable_pred p]
+  {f : {a : α // p a} → β} {f' : {a : α // ¬ p a} → β}
+  (hf : function.injective f) (hf' : function.injective f')
```

Comment:
Since we're already in the `function` namespace, you can do `hf : injective f` instead, and similar for `hf'` and the result

Reply:
Although note that above I suggestion moving to the `disjoint` namespace, which if you decide is a good idea would cause this comment not to apply.